source/Nuke.Common/Tools/CloudFoundry/CloudFoundryTasks.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 Maintainers of NUKE.
+// Distributed under the MIT License.
+// https://github.com/nuke-build/nuke/blob/master/LICENSE
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using ICSharpCode.SharpZipLib.GZip;
+using ICSharpCode.SharpZipLib.Tar;
+using Nuke.Common.Tooling;
+using static Nuke.Common.IO.HttpTasks;
+using static Nuke.Common.IO.FileSystemTasks;
+
+namespace Nuke.Common.Tools.CloudFoundry
+{
+    public static partial class CloudFoundryTasks
+    {
+
+        public static string GetToolPath()
+        {
+            try
+            {
+                return ToolPathResolver.TryGetEnvironmentExecutable("CLOUDFOUNDRY_EXE") ??
+                             ToolPathResolver.GetPathExecutable("cf");
+            }
+            catch
+            {
+            }
+            
+            var cliDir = NukeBuild.TemporaryDirectory / "cf-cli";
+            var binary = cliDir / (Environment.OSVersion.Platform == PlatformID.Win32NT ? "cf.exe" : "cf");
+            if (FileExists(binary))
+                return binary;
+
+            string releaseName;
```

Comment:
Personally I don't like that the executable is downloaded and installed automatically if not available. AFAIK we never do auto installing (besides dotnet). So I would prefer an exception and probably an install task.

What do you think about that @matkoch?